app/services/user_event_creator.rb

```diff
@@ -23,29 +29,48 @@ def create_out_of_band_user_event(event_type)
     create_event_for_device(event_type: event_type, user: current_user, device: nil)
   end
 
-  def create_user_event_with_disavowal(event_type, user = current_user)
-    event = create_user_event(event_type, user)
-    EventDisavowal::GenerateDisavowalToken.new(event).call
-    event
+  def create_user_event_with_disavowal(event_type, user = current_user, device = nil)
+    disavowal_token = SecureRandom.urlsafe_base64(32)
+    if device
+      create_event_for_existing_device(
+        event_type: event_type, user: user, device: device,
+        disavowal_token: disavowal_token
+      )
+    else
+      create_user_event(event_type, user, disavowal_token)
+    end
   end
 
   private
 
-  def create_event_for_existing_device(event_type:, user:, device:)
+  def create_event_for_existing_device(event_type:, user:, device:, disavowal_token:)
     device.update_last_used_ip(request.remote_ip)
-    create_event_for_device(event_type: event_type, user: user, device: device)
+    create_event_for_device(
+      event_type: event_type, user: user, device: device,
+      disavowal_token: disavowal_token
+    )
   end
 
-  def create_event_for_new_device(event_type:, user:)
+  def create_event_for_new_device(event_type:, user:, disavowal_token:)
     user_has_multiple_devices = UserDecorator.new(user).devices?
 
     device = create_device_for_user(user)
-    event = create_event_for_device(device: device, event_type: event_type, user: user)
-
-    return event unless user_has_multiple_devices
-
-    send_new_device_notification(user: user, event: event, device: device)
-    event
+    if user_has_multiple_devices && disavowal_token.nil?
+      event = create_user_event_with_disavowal(
+        event_type, user,
+        device
+      )
```

Comment:
I'd say we go with "all arguments on different lines, or all arguments on the same line"
```suggestion
      event = create_user_event_with_disavowal(
        event_type, user, device,
      )
```